libraries/aws/provisioning/include/aws_iot_provisioning.h

```diff
@@ -115,12 +116,59 @@ AwsIotProvisioningError_t AwsIotProvisioning_Init( uint32_t mqttTimeout );
  *   #AWS_IOT_PROVISIONING_INTERNAL_FAILURE, if any there are operation failures internal to the library.
  */
 /* @[declare_provisioning_createkeysandcertificate] */
-AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttConnection_t provisioningConnection,
+AwsIotProvisioningError_t AwsIotProvisioning_CreateKeysAndCertificate( IotMqttConnection_t connection,
                                                                        uint32_t flags,
                                                                        uint32_t timeoutMs,
                                                                        const AwsIotProvisioningCreateKeysAndCertificateCallbackInfo_t * pResponseCallback );
 /* @[declare_provisioning_createkeysandcertificate] */
 
+/**
+ * @brief Requests the AWS IoT Core service for a certificate by sending a Certificate-Signing Request, and
+ * invokes the provided user-defined response handler with the response it receives from the server.
+ *
+ * @note It is advised to use a shared MQTT connection to AWS IoT Core across all API functions.
+ *
+ * @warning This function is NOT thread-safe. Concurrent calls to the library API functions can result in undefined
```

Comment:
The non-thread-safe character of the library is a deliberate change I made recently (in [this PR](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/823) ) to simplify the internal implementation by removing the use of a mutex and an atomic reference counter. 
As the Foundry workflow requires a sequential invocation of these APIs, providing thread-safe APIs was just adding performance overhead and complexity which was unnecessary.

Reply:
If the API operates on a context structure that is provided as an explicit input to each call, and it's the responsibility of the caller to protect the sharing of that context, then this API should otherwise be implemented in a way that is thread safe (and commented as such).

Reply:
No context input is provided by the caller for any of the APIs that is shared between them. It is the internal state object of the library that is shared by each of the API functions, which is not protected for concurrent access and modification (as the Foundry provisioning workflow is ordered in the use of its APIs)

Reply:
In other words, the existing library already depends on global state?

Reply:
Yes.